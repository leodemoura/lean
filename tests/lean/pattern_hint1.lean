constants f g : nat → Prop

definition foo₁ [forward] : ∀ x, f x ∧ g x            := sorry
definition foo₂ [forward] : ∀ x, (: f x :) ∧ g x      := sorry
definition foo₃ [forward] : ∀ x, (: f (id x) :) ∧ g x := sorry

print foo₁
print foo₂
print foo₃ -- id is unfolded
