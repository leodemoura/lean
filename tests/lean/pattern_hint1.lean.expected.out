definition foo₁ [forward] : ∀ (x : ℕ), f x ∧ g x :=
sorry
(multi-)patterns:
?M_1 : ℕ
{f ?M_1}
{g ?M_1}
definition foo₂ [forward] : ∀ (x : ℕ), (:f x:) ∧ g x :=
sorry
(multi-)patterns:
?M_1 : ℕ
{f ?M_1}
definition foo₃ [forward] : ∀ (x : ℕ), (:f (id x):) ∧ g x :=
sorry
(multi-)patterns:
?M_1 : ℕ
{f ?M_1}
